src/azul/service/responseobjects/elastic_request_builder.py

```diff
@@ -552,6 +554,15 @@ def transform_request(self,
         if entity_type == 'projects':  # Add project summaries to each project hit
             self.add_project_summaries(final_response['hits'], es_response)
 
+        if include_file_urls:
+            try:
+                for h in final_response['hits']:
+                    for data_file in h['files']:
+                        data_file['url'] = f"{config.dss_endpoint}/files/{data_file['uuid']}" \
+                                           f"?replica=aws&version={data_file['version']}"
+            except KeyError as e:
+                logging.error('Unable to add file url. Due to the following exception:')
```

Comment:
Please don't write these blanket `except` clauses. I can count five dictionary lookups in the body of the `try:`, each one could cause a `KeyError`. Correct me if I am wrong but of neither of them should be swallowed like this because they would all constitute an error that needs to be surfaced very explicitly by returning a non-200 response. The only thing that a blanket clause like this achieves, is communicate that we are unsure about our invariants.

Reply:
Also, please take a look at how to log an exception: See `exc_info` in https://docs.python.org/3/library/logging.html#logging.debug

Reply:
So should I remove the try-except and allow the `KeyError` to resurface as a 500 error?

Reply:
Yes, unless you can show that a KeyError is expected to occur, where and under which circumstances. In that case the enclosing `try:` block should only contain a single statement with a single dictionary lookup.